lua/entities/magicbutton.lua

```diff
@@ -305,57 +325,321 @@ local function MagicOutcomeBountyAndPrize(ply)
+        weight = 4,
+    },
+    {
+        func = MagicOutcomeKleinerSlur,
+        message = "%s",
+        weight = 3
+    },
+    {
+        func = MagicOutcomeOverlay,
+        message = "and had their screen fucked up %s ;billhead;",
+        weight = 2
+    },
+    { 
+        func = MagicOutcomeSpawnObject,
+        message = "and it spawned %s",
+        weight = 2
     },
-    -- { --     func = MagicOutcomeKleinerFanclub, --     message = "and won the attention of every kleiner on the map!", -- }, -- { --     func = function(ply) KLEINER_BULLIES[ply:SteamID()] = 5000 return "" end, --     message = "and said an anti-kleiner slur! watch out!", -- },
     {
         func = function(ply, button)
             ply:SetPos(button:FindSuitableCastOrigin().StartPos)
 
+            return table.Random({";blackwhat;", "", "", ""})
+        end,
+        message = "and teleported somewhere mysterious! %s",
+        weight = 2
+    },
+    {
+        func = MagicOutcomeBountyAll,
+        message = "and [red]increased everyone's bounty by %s points! ;dougie;",
+        weight = 1
+    },
+    {
+        func = MagicOutcomeButtonSpawn,
+        message = "and spawned [rainbow2];weewoo;another button;weewoo;[fbc], which appeared somewhere in the location:[white] %s",
+        weight = 2
+    },
+    {
+        func = MagicOutcomeExplode,
+        message = "it exploded haha ;crazy;",
+        weight = 3
+    },
+    {
+        func = function(ply, button)
+            ply:EmitSound("physics/flesh/flesh_strider_impact_bullet1.wav")
+            local eang = ply:EyeAngles()
+            eang.roll = eang.roll + 15
+            ply:SetEyeAngles(eang)
+            local bone = ply:LookupBone("ValveBiped.Bip01_Head1") or ply:LookupBone("LRigScull")
+            local bang = ply:GetManipulateBoneAngles(bone)
+            bang.pitch = -eang.roll
+            ply:ManipulateBoneAngles(bone, bang)
+
             return ""
         end,
-        message = "and teleported somewhere mysterious!",
+        message = "and got a sore neck",
+        weight = 3
+    },
+    {
+        func = function(ply, button)
+            return table.Random({";baby;", ";bad;", ";biggestloser;", ";concern;", ";bartcry;", ";bazinga;", ";boohoo;", ";chungus;", ";eating;"})
+        end,
+        message = "but nothing happened! %s",
+        weight = 2
     },
     {
-        func = function(ply, button) return "" end,
-        message = "but nothing happened!",
+        func = function(ply, button)
+            if (OpenAPresent) then return OpenAPresent(ply, button:GetPos()) end
+        end,
+        message = "and got [white]%s[fbc]! ;alien;",
+        weight = 8
     }
 }
 
-for i = 1, 3 do
-    table.insert(MagicButtonOutcomes, {
-        func = function(ply, button) return OpenAPresent(ply, button:GetPos()) end,
-        message = "and got %s!",
-    })
-end
-
 --NOTE: On these functions, please return a string if there was a success, and nil if there was not.
```

Comment:
If this is your defined interface, don't return a number from those other functions, do the comma format within them

Reply:
you're right, i'lll change this